### PR TITLE
.github/workflows/coverity-scan.yaml: add github action for Coverity

### PR DESCRIPTION
**Pull request by alexmurray**
Since the scan.coverity.com API throttles submissions, run weekly for now.

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by alexmurray** [comment]
This can be seen running successfully in my fork at https://github.com/alexmurray/snapd/actions/runs/11439748250

File: .github/workflows/coverity-scan.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        fetch-depth: 0
+        # NOTE: checkout the code in a fixed location, even for forks, as this
+        # is relevant for go's import system.
+        path: ./src/github.com/snapcore/snapd
+    - name: Download Debian dependencies
+      run: |
+          sudo apt clean
+          sudo apt update
+          sudo apt build-dep -y "${{ github.workspace }}/src/github.com/snapcore/snapd"
+    - name: Install the go version
```
Comment (zyga) [inline .github/workflows/coverity-scan.yaml:42]: Do we need go for this test? Perhaps we could just skip this part?

**Review by zyga — COMMENTED** [review]
Thanks Alex! This looks like a nice addition. I will sync with you separately to make sure I can see any failures this shows up. I left a question about using go separately from what is provided in build-dependencies and if this is needed at all, given that covertity looks at snap-confine.

**Comment by alexmurray** [comment]
Thanks @zyga - good point - I was originally thinking that we might want to expand this to run Coverity against the entirety of snapd hence why I had the Go bits, but you make a good point. So I removed all the Go parts and have focused this to just snap-confine. In the future we can look at expanding it perhaps to include all of snapd via a separate project on scan.coverity.com.

Also  you should have access to the results already at https://scan.coverity.com/projects/snap-confine - I added you to the project back in May.

**Review by zyga — APPROVED** [review]

**Comment by alexmurray** [comment]
@sergiocazzolato I see you added the nightly tics workflow - but in that I can't see any reference to an `environment` from which to read the `secrets` from - in this PR I assume there is an environment called `normal` where the associated `COVERITY_SCAN_TOKEN` secret would live in the Github repo configuration but any advice you could give would be appreciated. Thanks.

**Comment by alexmurray** [comment]
@sergiocazzolato note you need to configure the token in the secrets for the repo as currently the job will fail without it - see https://github.com/canonical/snapd/actions/runs/11618548287/job/32356180152 for a manually triggered run which demonstrates the failure. I have invited you to the project in scan.coverity.com so you should be able to see the token there to configure in the snapd repo settings.

**Pull request merged**